File: src/main/java/com/kasisoft/libs/common/annotation/Specifications.java
package com.kasisoft.libs.common.annotation;

import java.lang.annotation.*;

/**
 * Container for multiple {@link Specification} elements.
 *
 * @author daniel.kasmeroglu@kasisoft.com
 */
@Retention(RetentionPolicy.CLASS)
@Target({ElementType.TYPE, ElementType.METHOD, ElementType.CONSTRUCTOR})
public @interface Specifications {

    /**
     * Returns a list of {@link Specification} elements.
     *
     * @return A list of {@link Specification} elements.
     */
    Specification[] value();

} /* ENDANNOTATION */

File: src/main/kotlin/com/kasisoft/libs/common/annotation/Specifications.kt
package com.kasisoft.libs.common.annotation

/**
 * Container for multiple {@link Specification} elements.
 *
 * @author daniel.kasmeroglu@kasisoft.com
 */
@Retention(AnnotationRetention.BINARY)
@Target(AnnotationTarget.CLASS, AnnotationTarget.FUNCTION, AnnotationTarget.CONSTRUCTOR)
annotation class Specifications(

    /**
     * Returns a list of {@link Specification} elements.
     *
     * @return A list of {@link Specification} elements.
     */
    val value: Array<Specification>

) /* ENDANNOTATION */
